tools/advisor/advisor/db_benchmark_client.py

```diff
@@ -0,0 +1,406 @@
+from abc import ABC, abstractmethod
+from advisor.db_log_parser import DataSource, DatabaseLogs, NO_FAM
+from advisor.db_options_parser import DatabaseOptions
+from advisor.db_stats_fetcher import (
+    LogStatsParser, OdsStatsFetcher, DatabasePerfContext
+)
+import glob
+import os
+import re
+import shutil
+import subprocess
+import time
+
+
+'''
+NOTE: This is not thread-safe, because the output file is simply overwritten.
+'''
+
+
+class BenchmarkRunner(ABC):
+    @staticmethod
+    @abstractmethod
+    def is_metric_better(new_metric, old_metric):
+        pass
+
+    @abstractmethod
+    def run_experiment(self):
+        # should return a list of DataSource objects
+        pass
+
+
+class DBBenchRunner(BenchmarkRunner):
+    OUTPUT_FILE = "temp/dbbench_out.tmp"
+    ERROR_FILE = "temp/dbbench_err.tmp"
+    DB_PATH = "DB path"
+    THROUGHPUT = "ops/sec"
+    PERF_CON = "db perf context"
+
+    # Map from Rocksdb option to its corresponding db_bench command-line arg
+    OPTION_CMD_LINE_FLAG = {
```

Comment:
This is for the case when options need to be given through the command-line instead of the options file, some options have a different name / value when given as command-line args. Such options are added to this map.

Reply:
Can you add this explanation to the inline comment?